source/Engine.cpp

```diff
@@ -1336,19 +1336,21 @@ void Engine::EnterSystem()
 				CreateWeather(hazard, stellar.Position());
 	}
 
-	for(const auto &raidFleet : system->GetGovernment()->RaidFleets())
-	{
-		double attraction = player.RaidFleetAttraction(raidFleet, system);
-		if(attraction > 0.)
-			for(int i = 0; i < 10; ++i)
-				if(Random::Real() < attraction)
-				{
-					raidFleet.GetFleet()->Place(*system, newShips);
-					Messages::Add("Your fleet has attracted the interest of a "
-							+ raidFleet.GetFleet()->GetGovernment()->GetName() + " raiding party.",
-							Messages::Importance::Highest);
-				}
-	}
+	if(!system->NoRaids())
+		for(const auto &raidFleet : !system->RaidFleets().empty() ?
+				system->RaidFleets() : system->GetGovernment()->RaidFleets())
```

Comment:
Could we not have this `!system->RaidFleets().empty() ? system->RaidFleets() : system->GetGovernment()->RaidFleets()` code be a part of `System::RaidFleets()`? The system has the pointer to its government, and so the system could do this check of which vector of fleets to return.
```suggestion
		for(const auto &raidFleet : system->RaidFleets())
```

Reply:
good point, that simplifies it in both places